lib/DxilPIXPasses/PixPassHelpers.cpp

```diff
@@ -450,6 +450,56 @@ void ReplaceAllUsesOfInstructionWithNewValueAndDeleteInstruction(
   delete Instr;
 }
 
+unsigned int FindOrAddSV_Position(hlsl::DxilModule &DM,
+                                  llvm::StringRef UpStream) {
+  hlsl::DxilSignature &InputSignature = DM.GetInputSignature();
+  auto &InputElements = InputSignature.GetElements();
+
+  auto Existing_SV_Position =
+      std::find_if(InputElements.begin(), InputElements.end(),
+                   [](const std::unique_ptr<DxilSignatureElement> &Element) {
+                     return Element->GetSemantic()->GetKind() ==
+                            hlsl::DXIL::SemanticKind::Position;
+                   });
+
+  // SV_Position, if present, has to have full mask, so we needn't worry
+  // about the shader having selected components that don't include x or y.
+  // If not present, we add it.
+  if (Existing_SV_Position == InputElements.end()) {
+
+    unsigned int Index = static_cast<unsigned int>(InputElements.size());
+
+    unsigned int StartRow = Index;
```

Comment:
`StartRow` should never just take the Index.  Is this path actually used, or is it always overridden by `UpStream` read below?  Even if we rely on it always being overridden, we should fail here somehow if not.  If it's possible you want to just add one without parsing from `UpStream`, then at least look for the next available row in the signature, rather than just taking the next signature element index.